### PR TITLE
Additions to the norwegian NKG2020 transformation

### PR DESCRIPTION
**Pull request by himsve**
The pull request is related and depended on pull request #https://github.com/OSGeo/PROJ-data/pull/58

### CODE REVIEW

**Comment by kbevers** [comment]
Please add a few test cases to https://github.com/OSGeo/PROJ/blob/master/test/gie/nkg.gie. The tests aren't checked with `make check` (because of large grids not in the main distribution) but some verification is still nice to have. I don't expect any issues now but with tests we can prevent potential future problems.

File: data/sql/nkg.sql [hunk 1]
```diff
@@ -693,7 +693,75 @@ INSERT INTO "usage" VALUES (
...
+    accuracy,
+    operation_version,
+    deprecated
+)
+VALUES(
+    'NKG', 'NKG_ETRF14_ETRF93_2000', -- object auth+code
+    'NKG_ETRF14 to ETRF93@2000.0', -- name
+    'Transformation from NKG_ETRF14 to ETRF93, at transformation reference epoch 2000.0', -- description / remark
+    'PROJ', 'PROJString', 
+    '+proj=xyzgridshift +grids=no_kv_NKGETRF14_EPSG7922_2000.tif',
```
Comment (rouault) [inline data/sql/nkg.sql:750]: that grid should likely be registered in data/sql/grid_alternatives.sql

File: data/sql/nkg.sql [hunk 2]
```diff
@@ -2056,6 +2124,116 @@ INSERT INTO "supersession" VALUES (
...
+    accuracy,
+    operation_version,
+    deprecated
+)
+VALUES(
+    'NKG', 'NO_2020_INTRAPLATE',        -- object auth+code
+    'ETRF93@2000.0 to ETRF93@1995.0',   -- name
+    NULL,                               -- description
+    'PROJ', 'PROJString',               -- method auth+cod
+    '+proj=deformation +dt=-5 +grids=eur_nkg_nkgrf17vel.tif',
```
Comment (rouault) [inline data/sql/nkg.sql:2152]: that one too

File: data/sql/nkg.sql (hunk 1 above)
Comment (himsve) [reply data/sql/nkg.sql:750]: @kbevers Should we add NKG resource files into grid_alternatives.sql?

Comment (kbevers) [reply data/sql/nkg.sql:750]: Doesn't hurt to do so. I am not entirely sure what is gained by doing it, as far as I know everything works as intended without it being there.

Comment (rouault) [reply data/sql/nkg.sql:750]: > I am not entirely sure what is gained by doing it,

The grid would then be referenced as available from the CDN

Comment (himsve) [reply data/sql/nkg.sql:750]: The NKG velocity grids 'eur_nkg_nkgrf03vel_realigned.tif' and 'eur_nkg_nkgrf17vel.tif' are not yet added to grid_alternatives. I tried, but I could not find any named 'proj_method' definition for geodetic velocity grids.

Comment (rouault) [reply data/sql/nkg.sql:750]: > I tried, but I could not find any named 'proj_method' definition for geodetic velocity grids.

You may need to add 'velocity_grid' in the enumeration of the constraint check_grid_alternatives_proj_method of the CREATE TABLE grid_alternatives definition in proj_db_table_defs.sql

File: data/sql/grid_alternatives.sql [hunk 3]
```diff
@@ -116,6 +116,10 @@ VALUES
 ('EGM08_REDNAP.txt','es_ign_egm08-rednap.tif',NULL,'GTiff','geoid_like',0,NULL,'https://cdn.proj.org/es_ign_egm08-rednap.tif',1,1,NULL),
 ('EGM08_REDNAP_Canarias.txt','es_ign_egm08-rednap-canarias.tif',NULL,'GTiff','geoid_like',0,NULL,'https://cdn.proj.org/es_ign_egm08-rednap-canarias.tif',1,1,NULL),
 
+-- eur_nkg - Nordic Geodetic Commission
+('NOT-YET-IN-GRID-TRANSFORMATION-eur_nkg_nkgrf03vel_realigned.tif','eur_nkg_nkgrf03vel_realigned.tif',NULL,'GTiff','velocity_grid',0,NULL,'https://cdn.proj.org/eur_nkg_nkgrf03vel_realigned.tif',1,1,NULL),
```
Comment (rouault) [inline data/sql/grid_alternatives.sql]: The NOT-YET-IN-GRID-TRANSFORMATION- prefix looks inappropriate.

Comment (himsve) [reply data/sql/grid_alternatives.sql]: I'm agreed. But without the NOT-YET-IN-GRID-TRANSFORMATION-prefix proj.db build fails (raised by grid_alternatives_insert_trigger). grid_alternatives_insert_trigger should may be triggered in the end of the proj.db build?

Comment (rouault) [reply data/sql/grid_alternatives.sql]: ah, I believe nkg.sql should be moved after ignf.sql and before grid_alternatives.sql in data/Makefile.am  and data/sql_filelist.cmake

Comment (himsve) [reply data/sql/grid_alternatives.sql]: Okey. I will test if that works.

**Pull request merged**